src/poetry/console/commands/plugin/remove.py

```diff
@@ -44,24 +47,69 @@ def handle(self) -> int:
             os.getenv("POETRY_HOME") if os.getenv("POETRY_HOME") else system_env.path
         )
 
-        # From this point forward, all the logic will be deferred to
-        # the remove command, by using the global `pyproject.toml` file.
-        application = cast(Application, self.application)
-        remove_command: RemoveCommand = cast(RemoveCommand, application.find("remove"))
-        # We won't go through the event dispatching done by the application
-        # so we need to configure the command manually
-        remove_command.set_poetry(Factory().create_poetry(env_dir))
-        remove_command.set_env(system_env)
-        application._configure_installer(remove_command, self._io)
-
-        argv = ["remove"] + plugins
-        if self.option("dry-run"):
-            argv.append("--dry-run")
-
-        return remove_command.run(
-            IO(
-                StringInput(" ".join(argv)),
-                self._io.output,
-                self._io.error_output,
+        existing_plugins = {}
+        if env_dir.joinpath("plugins.toml").exists():
+            existing_plugins = tomlkit.loads(
+                env_dir.joinpath("plugins.toml").read_text(encoding="utf-8")
             )
+
+        root_package = self.create_env_package(system_env, existing_plugins)
+
+        entrypoints = self.get_plugin_entry_points()
+
+        removed_plugins = []
+        for plugin in plugins:
+            plugin = canonicalize_name(plugin)
+            is_plugin = False
+            installed = False
+            for entrypoint in entrypoints:
+                if canonicalize_name(entrypoint.distro.name) == plugin:
+                    is_plugin = True
+                    break
+
+            for i, dependency in enumerate(root_package.requires):
+                if dependency.name == plugin:
+                    installed = True
+
+                    break
+
+            if not installed:
+                self.line_error(f"<warning>Plugin {plugin} is not installed.</warning>")
+
+                continue
+
+            if not is_plugin:
+                self.line_error(
+                    f"<warning>The package {plugin} is not a plugin.</<warning>"
+                )
+                continue
+
+            if plugin in existing_plugins:
+                del existing_plugins[plugin]
+
+            removed_plugins.append(plugin)
+
+        if not removed_plugins:
+            return 1
+
+        _root_package = root_package
```

Comment:
This shuffle is a bit gross -- can we clean up the variable names to be better here?